examples/reduction/examples/matmul.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+
+import cunumeric as np
+from reduction import (
+    matmul,
+    multiply,
+    print_store,
+    sum_over_axis,
+    to_cunumeric_array,
+    user_context,
+)
+
+import legate.core.types as ty
+
+
+def test(m: int, n: int, k: int, print_stores: bool, matmul_only: bool):
+    # Generate inputs using cuNumeric
+    rhs1 = user_context.create_store(ty.int64, (m, k))
+    rhs2 = user_context.create_store(ty.int64, (k, n))
+    tmp = np.arange(m * k).reshape(m, k)
+    to_cunumeric_array(rhs1)[:] = tmp
```

Comment:
this way of creating a temp array and then assigning through a slice via a cunumeric array was a bit odd to me - and might look a bit funny in a tutorial.

If we're going to depend on cunumeric, I would just create it as a cunumeric array and add a helper function `get_cunumeric_store`?